scripts/run-benchmarks.py

```diff
@@ -0,0 +1,185 @@
+import subprocess
+import json
+import os
+import re
+
+import click
+import cpuinfo
+
+BASE_REPO_URL = 'https://github.com/lumol-org/lumol.git'
+
+INDIVIDUAL_BENCH_TEMPLATE = """
+<details><summary>%s %s</summary>
+  <p>\n\n
+\n\n
+```bash\n
+%s\n\n
+```\n\n
+
+</p></details>
+"""
+
+COMPARISON_TEMPLATE = """
+<details><summary>%s %s</summary>
+  <p>\n\n
+\n\n
+```bash\n
+%s\n\n
+```\n\n
+
+</p></details>
+"""
+
+
+def request_api(endpoint, data=None):
+    url = 'https://api.github.com/repos/lumol-org/lumol' + endpoint
+    username = os.environ['LUMOL_GH_USERNAME']
+    token = os.environ['LUMOL_GH_TOKEN']
+
+    cmd = "curl -sS %s  -u '%s:%s'" % (url, username, token)
+    if data is not None:
+        cmd += " --data '%s'" % json.dumps(data)
+
+    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
+    return json.loads(out)
+
+
+def get_master_commit():
+    cmd = 'git log --oneline upstream/master -n 1'
+    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
+    h, _, title = out.rstrip().partition(' ')
+    return h, title
+
+
+def get_commit_descriptions(n_commits):
+    """
+    Get hash and title of the `n_commits` latest commits on the branch.
+
+    Also adds the commit at the HEAD of master in the end. If this
+    commit is met earlier, stops at this commit (the master commit
+    is guaranteed to be at the end of the result.
+
+    :param n_commits:
+    :return:
+    """
+    cmd = 'git log --oneline | head -n %s' % n_commits
+    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
+    master_h, master_title = get_master_commit()
+
+    descriptions = []
+    for line in out.split('\n'):
+        if line.rstrip() == '':
+            continue
+        h, _, title = line.partition(' ')
+        descriptions.append((h, title))
+        if h == master_h:
+            break
+
+    if descriptions[-1][0] != master_h:
+        descriptions.append((master_h, master_title))
+
+    return descriptions
+
+
+def setup_cloned_repo(pr_id):
+    response = request_api('/pulls/%s' % pr_id)
+    clone_url = response['head']['repo']['clone_url']
+    commit_id = response['head']['sha']
+
+    subprocess.call('git clone %s cloned_repo' % clone_url, shell=True)
```

Comment:
If you try to benchmark a PR initiated by me, you won't find my branch in your repo (because it comes from a fork). Are there cases where the bandwidth used to clone repetitively can be an issue ?